rts/lra/jaxrs/src/main/java/io/narayana/lra/filter/ServerLRAFilter.java

```diff
@@ -460,46 +575,149 @@ private boolean isJaxRsCancel(ContainerRequestContext requestContext, ContainerR
         return false;
     }
 
-    private URI startLRA(URI parentLRA, Method method, Long timeout) {
-        // timeout should already have been converted to milliseconds
-        String clientId = method.getDeclaringClass().getName() + "#" + method.getName();
+    // the request filter may perform multiple and in failure scenarios the LRA may be left in an ambiguous state:
+    // the following structure is used to track progress so that such failures can be reported in the response
+    // filter processing
+    private enum ProgressStep {
+        Left ("leave succeeded"),
+        LeaveFailed("leave failed"),
+        Started("start ok"),
+        StartFailed("start failed"),
+        Joined("join succeeded"),
+        JoinFailed("join failed"),
+        Ended("end ok"),
+        CloseFailed("close failed"),
+        CancelFailed("cancel failed");
+
+        final String status;
+
+        ProgressStep(final String status) {
+            this.status = status;
+        }
 
-        return lraClient.startLRA(parentLRA, clientId, timeout, ChronoUnit.MILLIS);
+        @Override
+        public String toString() {
+            return status;
+        }
     }
 
-    private void resumeTransaction(URI lraId) {
-        // nothing to do
+    // list of steps (both successful and unsuccesful) performed so far by the request and response filter
+    // and is used for error reporting
+    private static class Progress {
+        static EnumSet<ProgressStep> failures = EnumSet.of(
+                ProgressStep.LeaveFailed,
+                ProgressStep.StartFailed,
+                ProgressStep.JoinFailed,
+                ProgressStep.CloseFailed,
+                ProgressStep.CancelFailed);
+
+        ProgressStep progress;
+        String reason;
+
+        public Progress(ProgressStep progress, String reason) {
+            this.progress = progress;
+            this.reason = reason;
+        }
+
+        public boolean ok() {
```

Comment:
Maybe this would be better to be name with `success`? These this term seems to be used more in the sources here than `ok` to say that's all fine.

Reply:
It is a private method and there are lots of other words that mean the same thing.
If I change it to something you want then what about the next person who prefers something different.

Reply:
This is a trivial PR comment and is subjective. But if you're going to dig your feat in then I'll change it. I could be flippant and ask you for a coding style guide which passes your coding criteria, but I'll change it.